### ISSUE TITLE
Simplify visualization with less crossings and clear flow

### ISSUE THREAD

**Issue by hski-github** [hski-github/spotfire-sankey-diagram#5]
To align with colors from the color picker it was decided to not use Opacity like it is used in Sankey diagram of D3. But without opacity drawings are hidden by drawings in the foreground.

![image](https://user-images.githubusercontent.com/1921015/107694744-06428a00-6cb0-11eb-99d3-9b11401e52a9.png)

![image](https://user-images.githubusercontent.com/1921015/107684739-6aab1c80-6ca3-11eb-8024-b20a0f56ccb4.png)

![image](https://user-images.githubusercontent.com/1921015/107695156-7cdf8780-6cb0-11eb-9e4a-28f7631b68de.png)

**Comment by hski-github** [comment]
I am now sorting by value, that means big lines are in the background and small lines are in the foreground. This is a useful improvement I think.

![image](https://user-images.githubusercontent.com/1921015/107874574-cc919f00-6eba-11eb-95d5-00ad146bff01.png)

Still there are some optimizations useful for example if lines go in parallel from one bar to the other they should be drawn in the same layer (like blue in the screenshot below).

![image](https://user-images.githubusercontent.com/1921015/107874530-8f2d1180-6eba-11eb-9986-8fea50108947.png)

**Comment by hski-github** [comment]
Another example where optimization would be useful. Below you see unnecessary crossing of lines, if the order in the bars would be different e.g. yellow A could go straight from middle to right side but is crossing the blue line. And C on the right side if red and yellow would be changed, than layout would avoid unnecessary complexity. I guess at the end, there needs to be different sorting options.

![image](https://user-images.githubusercontent.com/1921015/107882348-06799a00-6ee9-11eb-80ae-49034fcbce60.png)

![image](https://user-images.githubusercontent.com/1921015/107882376-3163ee00-6ee9-11eb-904c-c1c5e3b1e100.png)

**Comment by hski-github** [comment]
Would be interested in seeing more examples with layout issues from other users to think about possible further layout optimization. Please post a screenshots here if you see potential for better Sankey layout.

**Comment by hski-github** [comment]
Now with recent code changes bar segments are sorted A > Z > Null, which gives more clarity for the visualization, still flow of rows need to be optimized to avoid unnecessary crossings.

**Comment by hski-github** [comment]
Sorting of bar.barsegments.rows is required to reduce number of unnecessary crossings and decrease of complexity. It should sort in the bar on where it goes in the next bar.

**Comment by hski-github** [comment]
With 0.9 some sorting optimization has been implemented. 

Now it is sorted by the order in the previous column.
![image](https://user-images.githubusercontent.com/1921015/117185134-ce903d80-add9-11eb-84b6-35aa077d5d0b.png)

Before it was like this
![image](https://user-images.githubusercontent.com/1921015/117185031-b0c2d880-add9-11eb-8fe6-a3931743bd81.png)

**Comment by hski-github** [comment]
Hello Filippo, could we move this conversation into a separat new issue? I guess this is a little different topic then what discussed in this issue here.

**Comment by f1li** [comment]
> Hello Filippo, could we move this conversation into a separat new issue? I guess this is a little different topic then what discussed in this issue here.

Yes of course Holger. Feel free to move it wherever you this it's most appropriate. By the way sorry If I commented here.

**Comment by hski-github** [comment]
With the recent improvements this issue can be closed. If new issues are reported by users then another round of improvements can start.
